plugins/c8y_log_plugin/src/main.rs

```diff
@@ -103,46 +114,7 @@ async fn run(
     loop {
         tokio::select! {
                 message = mqtt_client.received.next() => {
-                if let Some(message) = message {
-                    if is_c8y_bridge_up(&message) {
-                        plugin_config = read_log_config(config_file);
-                        let () = handle_dynamic_log_type_update(&plugin_config, mqtt_client).await?;
-                    }
-                    if let Ok(payload) = message.payload_str() {
-                        let result = match payload.split(',').next().unwrap_or_default() {
-                            "522" => {
-                                info!("Log request received: {payload}");
-                                // retrieve smartrest object from payload
-                                let maybe_smartrest_obj = SmartRestLogRequest::from_smartrest(payload);
-                                if let Ok(smartrest_obj) = maybe_smartrest_obj {
-                                    handle_logfile_request_operation(
-                                        &smartrest_obj,
-                                        &plugin_config,
-                                        mqtt_client,
-                                        http_client,
-                                    )
-                                    .await
-                                } else {
-                                    error!("Incorrect SmartREST payload: {}", payload);
-                                    Ok(())
-                                }
-                            }
-                            _ => {
-                                // Ignore operation messages not meant for this plugin
-                                Ok(())
-                            }
-                        };
-
-                        if let Err(err) = result {
-                            let error_message = format!("Handling of operation: '{}' failed with {}", payload, err);
-                            error!("{}", error_message);
-                        }
-                    }
-                }
-                else {
-                    // message is None and the connection has been closed
-                    return Ok(());
-                }
+               process_mqtt_messages(message, &plugin_config, mqtt_client, http_client, config_file).await?;
```

Comment:
By calling `process_mqtt_messages` only when there is actually a message, one can break the loop when there is no more messages.

```suggestion
               if let Some(message) = message {
                   process_mqtt_messages(message, &plugin_config, mqtt_client, http_client, config_file).await?;
               } else {
                    // message is None and the connection has been closed
                    return Ok(());
               }
```

Reply:
fixed